examples/rime-tsch/node.c

```diff
@@ -0,0 +1,100 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file is part of the Contiki operating system.
+ *
+ */
+
+/**
+ * \file
+ *         An example of Rime/TSCH
+ * \author
+ *         Simon Duquennoy <simon.duquennoy@inria.fr>
+ *
+ */
+
+#include <stdio.h>
+#include "contiki-conf.h"
+#include "net/netstack.h"
+#include "net/rime/rime.h"
+#include "net/mac/tsch/tsch.h"
+
+const linkaddr_t coordinator_addr =    { { 1, 0 } };
+const linkaddr_t sender_addr =         { { 2, 0 } };
+const linkaddr_t destination_addr =    { { 1, 0 } };
+
+/*---------------------------------------------------------------------------*/
+PROCESS(unicast_test_process, "Rime Node");
+AUTOSTART_PROCESSES(&unicast_test_process);
+
+/*---------------------------------------------------------------------------*/
+static void
+recv_uc(struct unicast_conn *c, const linkaddr_t *from)
+{
+  printf("App: unicast message received from %u.%u\n",
+   from->u8[0], from->u8[1]);
+}
+/*---------------------------------------------------------------------------*/
+static void
+sent_uc(struct unicast_conn *ptr, int status, int num_tx)
+{
+  printf("App: unicast message sent, status %u, num_tx %u\n",
+   status, num_tx);
+}
+
+static const struct unicast_callbacks unicast_callbacks = { recv_uc, sent_uc };
+static struct unicast_conn uc;
+
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(unicast_test_process, ev, data)
+{
+  PROCESS_BEGIN();
+
+  tsch_set_coordinator(linkaddr_cmp(&coordinator_addr, &linkaddr_node_addr));
+  NETSTACK_MAC.on();
+
+  unicast_open(&uc, 146, &unicast_callbacks);
+
+  while(1) {
+    static struct etimer et;
+
+    etimer_set(&et, CLOCK_SECOND);
+
+    PROCESS_WAIT_EVENT_UNTIL(etimer_expired(&et));
+
+    packetbuf_copyfrom("Hello", 5);
+
+    if(linkaddr_cmp(&sender_addr, &linkaddr_node_addr)) {
```

Comment:
You could change this line to:
`if(!linkaddr_cmp(&destination_addr, &linkaddr_node_addr)) {`
and then remove the definition of `sender_addr` on top. This way, if someone tries this example with multiple nodes, all the nodes will send unicast messages to the destination.